build/ci/github_actions/setup_ci_dependencies.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -xe
+set +o pipefail
+
+install_ci_dependencies (){
+  python -m pip install --upgrade pip
```

Comment:
always use specific versions, even for pip itself, to increase ci stability and determinism

Reply:
done